### DISCUSSION TITLE
Any Advice for Fixing an IDA_CONV_FAIL Error and IDA_TOO_MUCH_WORK Error?

### DISCUSSION THREAD

**Discussion by wass-cloud** [pybamm-team/PyBaMM#3169] [Q&A]
I've tried changing the mesh a bit, which worked last time this error occurred. If I understand what causes these errors I should be able to fix where my error is occurring, since I need my program to run with essentially no error to model a batteries lifetime over 600-700 hours.

**Comment by brosaplanella** [comment]
This error means that the solver failed to converge, which could be caused by many different factors, but there is no magic recipe that fixes them all. This [example](https://docs.pybamm.org/en/latest/source/examples/notebooks/solvers/speed-up-solver.html) shows some tricks on how to improve the solver.
